File: app/src/main/java/ds/vkplus/eventbus/EventBus.java
package ds.vkplus.eventbus;

import com.squareup.otto.Bus;

public class EventBus {

	static Bus instance;


	public static void post(Object o) {
		get().post(o);
	}


	public static void register(Object o) {
		get().register(o);
	}


	public static void unregister(Object o) {
		get().unregister(o);
	}


	private static Bus get() {
		if (instance == null)
			instance = new Bus();

		return instance;
	}

}

File: app/src/main/java/ds/vkplus/eventbus/EventBus.kt
package ds.vkplus.eventbus

import com.squareup.otto.Bus

object EventBus {

	private val instance: Bus by lazy { Bus() }

	public fun post(o: Any) = instance.post(o)
	public fun register(o: Any) {
		try {
			instance.register(o)
		} catch(e: Exception) {
		}
	}

	public fun unregister(o: Any) {
		try {
			instance.unregister(o)
		} catch(e: Exception) {
		}
	}
}
